### PR TITLE
LPD-2835 max terms count

### PR DESCRIPTION
**Pull request by gustavolimav**
https://liferay.atlassian.net/browse/LPD-2835

\cc @BryanEngler @AlmirFe

### CODE REVIEW

**Comment by gustavolimav** [comment]
ci:test:search

**Comment by gustavolimav** [comment]
ci:test:sf

**Comment by gustavolimav** [comment]
ci:stop

**Comment by gustavolimav** [comment]
ci:test:sf

**Comment by gustavolimav** [comment]
ci:test:search

**Comment by gustavolimav** [comment]
ci:help

**Comment by gustavolimav** [comment]
ci:test:search-opensearch2

File: modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/query/TermsQueryTranslatorImpl.java [hunk 1]
```diff
@@ -20,8 +23,36 @@ public class TermsQueryTranslatorImpl implements TermsQueryTranslator {
 
 	@Override
 	public QueryBuilder translate(TermsQuery termsQuery) {
-		return QueryBuilders.termsQuery(
-			termsQuery.getField(), termsQuery.getValues());
+		String field = termsQuery.getField();
+		int maxTermsCount = 65536;
+		String[] values = termsQuery.getValues();
```
Comment (BryanEngler) [inline modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/query/TermsQueryTranslatorImpl.java:28]: maybe rename values to terms

File: modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/query/TermsQueryTranslatorImpl.java [hunk 2]
```diff
@@ -20,8 +23,36 @@ public class TermsQueryTranslatorImpl implements TermsQueryTranslator {
...
+
+		if (values.length <= maxTermsCount) {
+			return new TermsQueryBuilder(field, values);
+		}
+
+		ArrayList<String> batchValues = new ArrayList<>();
```
Comment (BryanEngler) [inline modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/query/TermsQueryTranslatorImpl.java:34]: maybe rename batchValues to termsList

**Comment by BryanEngler** [comment]
do we have plans to add a test?

**Comment by gustavolimav** [comment]
@BryanEngler Almir was taking care of that, as tomorrow I will be on PTO Almir can be responsible to forward that PR with Tests

File: modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/query/TermsQueryTranslatorImpl.java (hunk 1 above)
Comment (gustavolimav) [reply modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/query/TermsQueryTranslatorImpl.java:28]: It was initially terms than I rename to values, but I agree terms sounds better.

**Comment by gustavolimav** [comment]
ci:reevaluate:1377401_16566

**Comment by gustavolimav** [comment]
ci:report:2564760023

**Comment by gustavolimav** [comment]
ci:test:sf

**Comment by gustavolimav** [comment]
Hey @peerkar,

I was wondering why we didn't use the same pattern as Elasticsearch of having a queryTranslator for each query type in OpenSearch2. Do you recall the reasoning behind this decision?

Currently, all the logic for query translation resides within OpenSearchQueryTranslator, whereas in Elasticsearch, we have dedicated translators for each query type.

Just curious to know if that was a specific motivation for that.

**Comment by peerkar** [comment]
> I was wondering why we didn't use the same pattern as Elasticsearch of having a queryTranslator for each query type in OpenSearch2. Do you recall the reasoning behind this decision?

A lot less service components and a lot less boilerplate code.

File: modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/query/TermsQueryTranslatorImpl.java [hunk 3]
```diff
@@ -20,8 +23,35 @@ public class TermsQueryTranslatorImpl implements TermsQueryTranslator {
...
+		String[] terms = termsQuery.getValues();
+
+		if (terms.length <= maxTermsCount) {
+			return new TermsQueryBuilder(field, terms);
+		}
+
+		ArrayList<String> termsList = new ArrayList<>();
+		BoolQueryBuilder boolQueryBuilder = new BoolQueryBuilder();
+
+		for (String value : terms) {
```
Comment (gustavolimav) [inline modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/query/TermsQueryTranslatorImpl.java:37]: rename `value` to `term`

File: modules/apps/portal-search-opensearch2/portal-search-opensearch2-impl/src/main/java/com/liferay/portal/search/opensearch2/internal/query/OpenSearchQueryTranslator.java [hunk 4]
```diff
@@ -670,396 +698,458 @@ public QueryVariant visit(MoreLikeThisQuery moreLikeThisQuery) {
...
+		if (terms.length <= maxTermsCount) {
+			return _getTermsQuery(termsQuery, field, terms);
+		}
 
-		SetterUtil.setNotNullFloat(builder::boost, termsQuery.getBoost());
+		ArrayList<String> termsList = new ArrayList<>();
+		BoolQuery.Builder boolQueryBuilder = QueryBuilders.bool();
 
-		builder.field(termsQuery.getField());
+		for (String value : terms) {
```
Comment (gustavolimav) [inline modules/apps/portal-search-opensearch2/portal-search-opensearch2-impl/src/main/java/com/liferay/portal/search/opensearch2/internal/query/OpenSearchQueryTranslator.java:1077]: Rename `value` to `term`

**Comment by gustavolimav** [comment]
ci:test:sf

**Comment by gustavolimav** [comment]
ci:test:sf

**Pull request closed without merging**